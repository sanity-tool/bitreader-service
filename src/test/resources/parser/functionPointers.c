void (*FPTR)();

extern void function();

static void takeAddress() {
    FPTR = function;
    FPTR = &function;
}

static void callFromPointer() {
    FPTR();
}