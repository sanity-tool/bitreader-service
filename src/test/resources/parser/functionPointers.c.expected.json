{
  "functions" : [ {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMStore",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        }, {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMStore",
        "operands" : [ {
          "kind" : "LLVMFunctionValueKind"
        }, {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMRet",
        "operands" : [ ]
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMCall",
        "operands" : [ {
          "kind" : "LLVMInstructionValueKind"
        } ]
      }, {
        "kind" : "LLVMRet",
        "operands" : [ ]
      } ]
    } ],
    "entryBlockIndex" : 0
  } ]
}